filter.php

```diff
@@ -274,6 +274,27 @@ private function get_embed_markup(string $linkhref, array $urls, array $options,
         $newtext = $videojs->embed($urls, $name, $width, $height, $embedoptions);
         // TODO: Deal with fallback link.
 
+        // We need to tweak the data-setup-lazy value so that it has html5->hls->enableLowInitialPlaylist enabled.
+        // This passes the enableLowInitialPlaylist value to the videojs JS, which fixes audio-only
+        // streams being played when the site bandwidth is too low.
+        // Regex and str_replace are used so that we don't have to touch the Moodle core videojs code.
+        $pattern = '/data-setup-lazy=["\']([^"\']+)["\']/i';
+        $matches = [];
+
+        if (preg_match($pattern, $newtext, $matches)) {
+            $originalvalue = $matches[1];
```

Comment:
This feels a bit magic, Can you add a comment on why matches[1] is what we want, as opposed to matches[0] etc. It just currently feels a bit opaque.

Reply:
Ah yeah it is a bit magic - matches[1] is just what was matched - will update with a doc tho

https://www.php.net/manual/en/function.preg-match.php

```
 matches

    If matches is provided, then it is filled with the results of search. $matches[0] will contain the text that matched the full pattern, $matches[1] will have the text that matched the first captured parenthesized subpattern, and so on.
```